tsl/src/compression/create.c

```diff
@@ -1113,13 +848,68 @@ tsl_process_compress_table(AlterTableCmd *cmd, Hypertable *ht,
 	}
 
 	/*add the constraints to the new compressed hypertable */
-	ht = ts_hypertable_get_by_id(ht->fd.id); /*reload updated info*/
-	ts_hypertable_clone_constraints_to_compressed(ht, constraint_list);
+	// ht = ts_hypertable_get_by_id(ht->fd.id); /*reload updated info*/
+	// ts_hypertable_clone_constraints_to_compressed(ht, constraint_list);
 
 	/* do not release any locks, will get released by xact end */
 	return true;
 }
 
+/*
+ * Verify uncompressed hypertable is compatible with conpression
+ */
+static void
+validate_hypertable_for_compression(Hypertable *ht)
+{
+	if (TS_HYPERTABLE_IS_INTERNAL_COMPRESSION_TABLE(ht))
+	{
+		ereport(ERROR,
+				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+				 errmsg("cannot compress internal compression hypertable")));
+	}
+
+	/*check row security settings for the table */
+	if (ts_has_row_security(ht->main_table_relid))
+		ereport(ERROR,
+				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+				 errmsg("compression cannot be used on table with row security")));
+
+	Relation rel = table_open(ht->main_table_relid, AccessShareLock);
+	TupleDesc tupdesc = rel->rd_att;
+
+	Size row_size = MAXALIGN(SizeofHeapTupleHeader);
+	row_size += 8;	/* sequence_num */
+	row_size += 4;	/* count */
+	row_size += 16; /* min/max */
```

Comment:
We should have a canonical place (header) where we define the metadata columns, including types and sizes. If we hard-code the information like this in every place it i used, it is easy to make a mistake.